internal/services/currency/currency_service.go

```diff
@@ -23,6 +24,11 @@ func (s *currencyService) Convert(ctx context.Context, params services.ConvertCu
 	if len(params.Target) == 0 {
 		return nil, fmt.Errorf("%w: no target currencies provided", services.InvalidArgumentErr)
 	}
+	for _, ccy := range entities.CurrenciesFromString(append(params.Target, params.Base)) {
+		if !ccy.IsSupported() {
```

Comment:
You should incapsulate this logic in CurrenciesFromString if you want do it like so.
So whenever  you try to convert currencies, you are making additional validation, so invalid currencies will not appear

Reply:
Replaced `CurrenciesFromString(data []string)` with `ValidateCurencies(data []string) error` and used it here